### PR TITLE
Feat/setup authentication

### LINKED ISSUE #25: [BE] Wire up Authentication to API
Instead of using Auth0, let's have metamask sign the request, and have backend verify it and generate a GWT token to authenticate subsequent requests from frontend.

### PR DESCRIPTION
**Pull request by AtelyPham**
* Setup `authentication`, `authorization`, `CRU` for `User` entity.
* Add role for `User` entity.
* Refactor and remove unused file and code.
* Refactor some `ResolveField`s to prevent cycle in `REST` responses.
* Setup `cors` and `exception-filter` for better error messages.

Thanks for previewing.

### CODE REVIEW

File: backend/src/common/middlewares/auth.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  next: NextFunction,
+) {
+  try {
+    const token =
+      req.headers["authorization"] || req.headers["Authorization"] || "";
+    const { address } = await Web3Token.verify(token);
+    req.currentUserAddress = address || null;
+
+    const wallet = await getMongoRepository(Wallet).findOne({ address });
+    const user = await getMongoRepository(UserEntity).findOne(wallet.userId);
```
Comment (thangle-s4v) [inline backend/src/common/middlewares/auth.middleware.ts:36]: can we add the check here when wallet is not null @AtelyPham ?

**Review by thangle-s4v — COMMENTED** [review]
Looks good em, exxept co cai small suggestion thoi

File: backend/src/common/middlewares/auth.middleware.ts (hunk 1 above)
Comment (AtelyPham) [reply backend/src/common/middlewares/auth.middleware.ts:36]: Ah a wallet always belongs to a user. So I skip that checking... Anyway, I will add the check for the code more robust :) Thanks anh <3

**Comment by AtelyPham** [comment]
Updated! Already added your suggestion ✅.

**Review by thangle-s4v — APPROVED** [review]
LGTM

**Comment by coleslaw** [comment]
let's merge =) @AtelyPham

**Pull request merged**